backend/src/main/java/codezap/template/domain/Template.java

```diff
@@ -65,16 +67,61 @@ public class Template extends SkipModifiedAtBaseTimeEntity {
     @Enumerated(EnumType.STRING)
     private Visibility visibility;
 
-    public Template(Member member, String title, String description, Category category, Visibility visibility) {
-        this(null, member, title, description, category, null, 0L, visibility);
+    @Column(nullable = false)
+    @ColumnDefault("0")
+    private Long thumbnailOrdinal;
+
+    @ElementCollection(fetch = FetchType.LAZY)
+    @OrderColumn(name = "ordinal")
+    @CollectionTable(name = "source_code", joinColumns = @JoinColumn(name = "template_id"))
+    private final List<SourceCode> sourceCodes = new ArrayList<>();
+
+    public Template(
+            Long id,
+            Member member,
+            String title,
+            String description,
+            Category category,
+            Long likesCount,
+            Visibility visibility,
+            Long thumbnailOrdinal,
+            List<SourceCode> sourceCodes
+    ) {
+        this.id = id;
+        this.member = member;
+        this.title = title;
+        this.description = description;
+        this.category = category;
+        this.likesCount = likesCount;
+        this.visibility = visibility;
+        this.thumbnailOrdinal = thumbnailOrdinal;
+        this.sourceCodes.addAll(sourceCodes);
+        validateSourceCodeCount(sourceCodes);
+    }
+
+    public Template(Member member, String title, String description, Category category, Visibility visibility, List<SourceCode> sourceCodes) {
+        this(null, member, title, description, category, 0L, visibility, 0L, sourceCodes);
     }
 
-    public void updateTemplate(String title, String description, Category category, Visibility visibility) {
+    public void updateTemplate(String title, String description, Category category, Visibility visibility, List<SourceCode> sourceCodes) {
         this.modifiedAt = LocalDateTime.now();
         this.title = title;
         this.description = description;
-        this.category = category;
         this.visibility = visibility;
+        this.category = category;
+        recreateSourceCodes(sourceCodes);
+    }
+
+    private void validateSourceCodeCount(List<SourceCode> sourceCodes) {
+        if (sourceCodes.isEmpty()) {
+            throw new CodeZapException(ErrorCode.INVALID_REQUEST, "소스 코드는 최소 1개 입력해야 합니다.");
+        }
+    }
+
+    private void recreateSourceCodes(List<SourceCode> sourceCodes) {
+        this.sourceCodes.clear();
+        this.sourceCodes.addAll(sourceCodes);
```

Comment:
질문! 

```suggestion
        this.sourceCodes = sourceCodes;
```

soureCodes에 final도 안붙어있겠다 그냥 이런 식으로 사용해도 될 거 같은데 지웠다가 다시 추가하는 형태로 구현한 이유가 궁금쓰

위에서 Template 생성할 때도 비슷하게 하는 것 같던데